static/js/controller/YouTubeSearchController.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+this.YouTubeSearchController = function($scope, $log, $timeout, socket) {
+  $scope.choosenTrack = null;
+
+  $scope.update = _.debounce(function() {
+    $scope.choosenTrack = null;
+    $.get(
+      "https://www.googleapis.com/youtube/v3/search",
+      {
+        q: $scope.searchTerm,
+        part: "snippet",
+        maxResults: 25,
+        type: "video",
+        videoCategoryId: "10", // Music
```

Comment:
Perhaps there are other restrictions we should set. The docs on the API console seem good. Haven't read through all of them yet though: https://developers.google.com/apis-explorer/#p/youtube/v3/youtube.search.list?q=surfing&maxResults=25&part=snippet